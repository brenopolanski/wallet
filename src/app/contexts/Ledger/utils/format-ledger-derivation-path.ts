import { LedgerDerivationScheme } from "app/contexts/Ledger/contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
