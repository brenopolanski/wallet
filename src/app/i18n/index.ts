import i18n from "i18next";
import { initReactI18next } from "react-i18next";

// Helpers
import { buildTranslations } from "./helpers";

const defaultNS = "translation";

const resources = {
	en: {
		translation: buildTranslations(),
	},
};

// eslint-disable-next-line
i18n.use(initReactI18next).init({
	defaultNS,
	lng: "en",
	ns: [defaultNS],
	resources,
});

export { defaultNS, i18n, resources };
