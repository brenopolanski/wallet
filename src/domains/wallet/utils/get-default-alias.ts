import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let sameCoinWalletsCount = sameCoinWallets.length || 1;

	while (profile.wallets().findByAlias(makeAlias(sameCoinWalletsCount))) {
		sameCoinWalletsCount++;
	}

	return makeAlias(sameCoinWalletsCount);
};
