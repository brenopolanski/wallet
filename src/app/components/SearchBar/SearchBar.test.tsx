/* eslint-disable @typescript-eslint/require-await */
import { waitFor } from "@testing-library/react";
import { translations } from "app/i18n/common/i18n";
import React from "react";
import { fireEvent, render, screen } from "utils/testing-library";

import { SearchBar } from "./SearchBar";

describe("SearchBar", () => {
	it("should render", () => {
		const { asFragment } = render(<SearchBar />);

		expect(screen.getByTestId("SearchBar")).toHaveTextContent(translations.SEARCH_BAR.FIND_IT);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render with children", () => {
		render(<SearchBar>I am a children</SearchBar>);

		expect(screen.getByText("I am a children")).toBeInTheDocument();
	});

	it("should call onSearch callback on button click", async () => {
		const onSearch = jest.fn();

		render(<SearchBar onSearch={onSearch} />);

		fireEvent.change(screen.getByTestId("Input"), {
			target: {
				value: "test query",
			},
		});

		fireEvent.click(screen.getByTestId("SearchBar__button"));

		await waitFor(() => expect(onSearch).toHaveBeenCalledWith("test query"));
	});
});
