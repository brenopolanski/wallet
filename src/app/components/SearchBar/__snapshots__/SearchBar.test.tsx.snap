// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render 1`] = `
<DocumentFragment>
  <div
    class="bg-theme-secondary-100 px-10 pt-8 pb-8"
    data-testid="SearchBar"
  >
    <div
      class="flex items-center py-6 px-10 rounded shadow-xl bg-theme-background"
    >
      <div
        class="sc-bdvvtL fCrhkg mr-8 w-4 text-theme-secondary-300"
        height="16"
        width="16"
      >
        <svg>
          magnifying-glass.svg
        </svg>
      </div>
      <div
        class="flex-1 pl-4 mr-4 border-l border-theme-secondary-300 dark:border-theme-secondary-800"
      >
        <div
          class="Input__InputWrapperStyled-sc-1i8sd1r-0 bjEupA"
        >
          <div
            class="relative flex flex-1 h-full"
          >
            <input
              class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
              data-testid="Input"
              placeholder="What would you like to search for?"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        class="OriginalButton__StyledButton-sc-o0fs2j-0 isLyLB my-1"
        data-testid="SearchBar__button"
        type="button"
      >
        <div
          class="flex items-center space-x-2"
        >
          <span
            class="px-2 text-md"
          >
            Find it
          </span>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
